09 Redux/src/reducers/memberReducer.ts

```diff
@@ -0,0 +1,83 @@
+import MemberEntity from "../api/memberEntity";
+import MemberAPI from "../api/memberAPI";
+import objectAssign = require('object-assign');
+import MemberFormErrors from "../validations/memberFormErrors"
+import MemberFormValidator from "../validations/memberFormValidator"
+
+
+let emptyMemberEntity = new MemberEntity();
+
+class MemberState  {
+  member : MemberEntity;
+  memberId : number;
+  errors : MemberFormErrors;
+  isValid : boolean;
+  saveCompleted : boolean;
+
+  public constructor()
+  {
+    this.member = new MemberEntity();
+    this.memberId = -1;
+    this.errors = new MemberFormErrors();
+    this.isValid = false;
+    this.saveCompleted = false;
+  }
+}
+
+
+// Just to show how combine reducers work, we have
+// divided into two reducers member load + member load/update/delete
+export default (state : MemberState = new MemberState(), action) => {
+  let newState : MemberState = null;
+
+  switch (action.type) {
+    case 'MEMBER_NEW':
```

Comment:
I think that rename to 'INITIALIZE_NEW_MEMBER' is more explicit.

What do you think?